draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -485,16 +485,29 @@ a Disclosure more than once.
 ### Enabling Holder Binding {#enabling_holder_binding}
 
 The Holder MAY add an optional JWT to prove Holder Binding to the Verifier.
-The precise contents of the JWT are out of scope of this specification.
-Usually, a `nonce` and `aud` claim are included to show that the proof is
-intended for the Verifier and to prevent replay attacks. How the `nonce` or
-other claims are obtained by the Holder is out of scope of this specification.
 
-Example Holder Binding JWT payload:
+#### Structure of the Holder Binding JWT {#hb-jwt}
+
+This section defines the contents of the Holder Binding JWT.
+
+The JWT MUST contain the following elements:
+  * in the JOSE header,
+    * `typ`: REQUIRED. MUST be `hb+jwt`, which explicitly types the Holder Binding JWT as recommended in Section 3.11 of [@!RFC8725].
+    * `alg`: REQUIRED. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry. MUST NOT be `none` or an identifier for a symmetric algorithm (MAC).
+    * `kid`: `kid`: CONDITIONAL. JOSE Header containing the key ID. MUST NOT be present if `jwk` or `x5c` is present.
+    * `jwk`: CONDITIONAL. JOSE Header containing the key material the new Credential shall be bound to. MUST NOT be present if `kid` or `x5c` is present.
+    * `x5c`: CONDITIONAL. JOSE Header containing a certificate or certificate chain corresponding to the key used to sign the JWT. This element MAY be used to convey a key attestation. In such a case, the actual key certificate will contain attributes related to the key properties. MUST NOT be present if `kid` or `jwk` is present.
+  * in the JWT body,
+    * `iss`: REQUIRED. The issuer of the Holder Binding JWT. How the value is represented is up to the protocol used and out of scope of this specification.
```

Comment:
I don't think issuer should be required. It's kinda implied by context to be the Holder. And doesn't really add anything to have it, that I can see. And there may not always be an obvious value to use for the iss value for the holder. Seems more likely to hurt than help.

Reply:
Oh and also the current examples of the Holder Binding JWT or its payload have iat, aud, and nonce. But not iss. Removing it here, where it's not necessary IMHO, would be easier than updating the code and examples. 

Also I suspect there's some reason Dr. Fett didn't have `iss` in the HB JWT payloads. Even if not normative examples.

Reply:
> Also I suspect there's some reason Dr. Fett didn't have `iss` in the HB JWT payloads. Even if not normative examples.

Essentially what you said above - (1) I'm not sure if there's always a defined value to put there and (2) I don't think it is necessary. Audience is really important, but I don't see a strict need for iss.

Reply:
```suggestion
```